addons/web/static/src/core/pager/pager.js

```diff
@@ -53,20 +54,30 @@ export class Pager extends Component {
      * @returns {boolean} true if there is only one page
      */
     get isSinglePage() {
-        return this.minimum === 1 && this.maximum === this.props.total;
+        return this.minimum === 1 && this.maximum === this.state.total;
+    }
+    /**
+     * Recompute total number of records, if we were at the limit of 10000
+     */
+    async updateCounter() {
+        if (this.state.total == 10000) {
+            this.state.total = await this.env.searchModel.getSearchCount();
+        }
     }
-
     /**
      * @param {-1 | 1} direction
      */
     navigate(direction) {
         let minimum = this.props.offset + this.props.limit * direction;
-        if (minimum >= this.props.total) {
+        if ((minimum+this.props.limit) >= this.state.total) {
+            this.updateCounter();
```

Comment:
probably need to add `await` here, so the rest of the code uses the updated value (so, the `navigate` function need to be `async`

Reply:
Not sure about that; the query could be slow (it happens when there are more than 10k records).

Even though it does a rerender, I kept it that way to not slow down the rest of the UI. The pager widget is very small, so the feeling in the UI was quite good. (and it's quite rare; it happens only when you browse after 10k records, and only the first time you pass the limit)